[Infra] Turn off asset details dashboard tab by default

jennypavlova opened the pull request:
## Summary

This PR changes the feature flag (`observability:enableInfrastructureAssetCustomDashboards`) because the dashboards tab should be hidden by default.

The flag should be technical preview (not beta): 

<img width="1920" alt="image" src="https://github.com/elastic/kibana/assets/14139027/4b630881-1b4d-41c3-bc6a-1d43436b339b">


## Testing
Inside the infrastructure settings page the `observability:enableInfrastructureAssetCustomDashboards` should be false by default: 

![image](https://github.com/elastic/kibana/assets/14139027/5ef5943b-3b96-47b1-926d-0bec0d7cab5f)

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `run` `docs-build` : Re-trigger the docs validation. (use unformatted text in the comment!)
</p>
</details>

bmorelli25 reviewed (commented):
It looks like this feature is in beta now? Can you pls update the docs. Here's the line. Just change `preview::` to `beta::`

https://github.com/elastic/kibana/blob/8.13/docs/management/advanced-options.asciidoc?plain=1#L467

bmorelli25 reviewed (approved):
I'll approve so you can merge after the doc update. Thanks!

jennypavlova commented:
@bmorelli25 good catch, I think it should be a technical preview and not beta in the config. 
Getting @roshan-elastic here to help us: The dashboards tab should be marked as a technical preview (not beta) right?
If yes I can change the beta label instead.

roshan-elastic commented:
> @bmorelli25 good catch, I think it should be a technical preview and not beta in the config. Getting @roshan-elastic here to help us: The dashboards tab should be marked as a technical preview (not beta) right? If yes I can change the beta label instead.

That's right @jennypavlova - good catch @bmorelli25 !

jennypavlova commented:
@roshan-elastic and @bmorelli25 Thank you both! I changed it to technical preview and updated the PR description, so now it shows it correctly also in the settings: 
<img width="1920" alt="image" src="https://github.com/elastic/kibana/assets/14139027/4b630881-1b4d-41c3-bc6a-1d43436b339b">

I will merge this once the build is ready.
